test/functional/p2p_eviction.py

```diff
@@ -122,6 +122,40 @@ def run_test(self):
         self.log.debug("{} protected peers: {}".format(len(protected_peers), protected_peers))
         assert evicted_peers[0] not in protected_peers
 
+        self.log.info("Test that whitebind inbounds get extra eviction power")
+        # Only allow 1 inbound connection, but set whitebind
+        self.restart_node(0, extra_args=['-maxconnections=12', '-whitebind=127.0.0.1:30201', '-whitebind=forceinbound@127.0.0.1:30202'])
+        self.log.debug("Connect 1 peer")
+        node.add_p2p_connection(P2PInterface())
+
+        # Create a peer that expects to be rejected
+        class RejectedPeer(P2PInterface):
+            def connection_lost(self, exc):
+                return
+
+        self.log.debug("Generic inbound gets rejected when full")
+        with node.assert_debug_log(["failed to find an eviction candidate - connection dropped (full)"]):
+            node.add_p2p_connection(RejectedPeer(), wait_for_verack=False)
+
+        self.log.debug("Default whitebind inbound gets rejected, even when full")
+        with node.assert_debug_log(["failed to find an eviction candidate - connection dropped (full)"]):
+            node.add_p2p_connection(RejectedPeer(), wait_for_verack=False, dstport=30201)
+
+        self.log.debug("ForceInbound whitebind inbound gets connected, even when full")
```

Comment:
I'm trying to figure out if
1. Local peers are protected in any special way (I don't think they are?)
2. Inbound onion connections are considered local (They aren't - inbound onions are identifiable because they bind to a different network port `onion_service_target_port`)

And then, I don't think all onion peers are protected per se but their latency is compensated for in the eviction process:

https://github.com/bitcoin/bitcoin/blob/c9f288244b8d183e09a917025922b99e3368ef78/src/node/eviction.cpp#L105-L112